gio/src/unix_fd_message.rs

```diff
@@ -0,0 +1,50 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::UnixFDMessage;
+use glib::object::IsA;
+use glib::translate::*;
+use std::{mem, ptr};
+
+#[cfg(unix)]
+use std::os::unix::io::{AsRawFd, RawFd};
+
+#[cfg(all(not(unix), feature = "dox"))]
+use socket::{AsRawFd, RawFd};
+
+pub trait UnixFDMessageExtManual: Sized {
+    #[doc(alias = "g_unix_fd_message_append")]
+    fn append_fd<T: AsRawFd>(&self, fd: T) -> Result<(), glib::Error>;
```

Comment:
`IntoRawFd` maybe? This needs to take ownership of the fds or otherwise somehow have a lifetime bound to the fds

Reply:
It calls `dup` to make a copy of the fd: https://docs.gtk.org/gio/method.UnixFDMessage.append_fd.html

This is the same behavior as `UnixFDListExtManual::append` which also takes `AsRawFd`.

Reply:
Ok, makes sense